model/textparse/protobufparse.go

```diff
@@ -526,67 +552,77 @@ func (p *ProtobufParser) Next() (Entry, error) {
 	return p.state, nil
 }
 
-func (p *ProtobufParser) updateMetricBytes() error {
-	b := p.metricBytes
-	b.Reset()
-	b.WriteString(p.getMagicName())
-	for _, lp := range p.mf.GetMetric()[p.metricPos].GetLabel() {
-		b.WriteByte(model.SeparatorByte)
-		n := lp.GetName()
-		if !model.LabelName(n).IsValid() {
-			return fmt.Errorf("invalid label name: %s", n)
-		}
-		b.WriteString(n)
-		b.WriteByte(model.SeparatorByte)
-		v := lp.GetValue()
-		if !utf8.ValidString(v) {
-			return fmt.Errorf("invalid label value: %s", v)
-		}
-		b.WriteString(v)
+// onSeriesOrHistogramUpdate updates internal state before returning
+// a series or histogram. It updates:
+// * p.lset (copy).
+// * p.seriesBytes (copy).
```

Comment:
What is meant by "copy" here?  `lset` ends up using a buffer in `p.builder`, and `seriesBytes` will re-use its own buffer.